tests_lib/common/custom_logger.py

```diff
@@ -1,49 +1,74 @@
-import logging
-import os
+import shutil
+
 from tests_lib.common.yaml_loaders import load_config
+import logging
+from pathlib import Path
 
 
 class CustomLogger(logging.Logger):
     """
     A custom logger class for managing test logs with extended functionality.
-
-    This class extends the base logging.Logger to provide preconfigured logging settings,
-    including a log file handler, a custom log format, and a utility method to add dividers
-    between log sessions for better readability.
     """
     def __init__(self, logger_name: str, log_file_name: str):
         """
-        Overrides the base class constructor to configure a custom logger.
-
-        Sets up a log file handler with a specified file name, applies a custom format, and
-        stores logs in the "/opt/project/logs" folder. The logger level is set to in config/common_config.yaml
+        Initializes a custom logger with a file handler and custom formatter.
 
         :param logger_name: Name of the logger instance.
-        :param log_file_name: Name of the log file where logs will be saved.
+        :param log_file_name: Name of the log file.
+        """
+        # Create a logger
+        super().__init__(logger_name, self._get_log_level())
+
+        # Set up log file path
+        self.log_file_path = self._get_log_file_path(log_file_name)
+
+        # Set up the file handler and add it to the logger
+        file_handler = self._create_file_handler()
+        self.addHandler(file_handler)
+
+    def _get_log_level(self) -> str:
+        """
+        Gets the log level from a config file.
+
+        :return: String representation of the level.
+        """
+        config = load_config("logger_config")
+        return config["log_level"]
+
+    def _get_log_file_path(self, log_file_name: str) -> Path:
+        """
+        Constructs the path to the log file.
+
+        :param log_file_name: Name of the log file.
+        :return: Full path to the log file.
         """
-        config = load_config("common_config")
-        super().__init__(logger_name, config["log_level"])
+        logs_folder = Path(__file__).resolve().parent.parent.parent / "logs"

```

Comment:
Please, move the log file path back to the config file, because it's better to have it configurable than hard-coded here. 
Your code will be broken in case custom_logger.py file is moved to another directory. Code with this path stored in config file is more flexible and reliable.